### DISCUSSION TITLE
Using python libraries in xonsh that need platform support

### DISCUSSION THREAD

**Discussion by nimatrueway** [xonsh/xonsh#4896] [Q&A] [answered]
As I understand xonsh has it's own isolated installation of Python, so I use the following command to install libraries that I intend to import in my xonsh scripts

```
@(sys.executable) -m pip install {package}
```

It's all fine, until you'd like to use libraries that need native/platform support such as `easygui` that requires `tk` nativelib and bindings:

```
sudo apt-get install python3-tk // ubuntu
brew install python-tk // macOS
```

Installing them like that did not help to make them available to `xonsh`'s specific python runtime as I'm still getting the error:

```
import tkinter
xonsh: For full traceback set: $XONSH_SHOW_TRACEBACK = True
ModuleNotFoundError: No module named '_tkinter'
```

but not on platform's python

```
> python3
Python 3.9.13 (main, May 24 2022, 21:28:31)
[Clang 13.1.6 (clang-1316.0.21.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import tkinter
>>>
```

Is there a way to install and leverage such libraries for `xsh` scripts?

**Comment by nimatrueway** [comment, marked as answer]
Ahh `brew install python-tk@3.10` did the trick. Somehow `/usr/local/bin/python3` on my machine was linked to `python3.9` but `xonsh` was using `python3.10`.

**Reply by bestlem** [reply]
What is the first line of xonsh? I don't think it has its own python
